.github/workflows/main.yaml

```diff
@@ -39,15 +43,18 @@ jobs:
           docker build . --tag $IMAGE  \
             --label "org.opencontainers.image.source=https://github.com/$GH_REPOSITORY"
           docker push $IMAGE
-          echo "IMAGE=$IMAGE" >> $GITHUB_OUTPUT
+
+          # Get image digest to pass on to deployment for pinning the image to that digest
+          IMAGE_DIGEST=$(docker inspect --format='{{index .RepoDigests 0}}' $IMAGE)
+          echo "IMAGE_DIGEST=$IMAGE_DIGEST" >> $GITHUB_OUTPUT
```

Comment:
Previously any successful main deployment set `image:latest`, which actually doesn't roll the k8s pods as the image tag didn't change. Align with the digest used in the PR flow.